lib/index.ts

```diff
@@ -17,14 +17,12 @@ function jwtDecode<T = JwtHeader>(
   options: JwtDecodeOptions & { header: true }
 ): T;
 function jwtDecode<T = JwtPayload>(token: string, options?: JwtDecodeOptions): T;
-function jwtDecode(
-  token: string,
-  options: JwtDecodeOptions = { header: false }
-) {
+function jwtDecode(token: string, options?: JwtDecodeOptions) {
```

Comment:
What is wrong with providing a default? I am not sure I follow.

Reply:
By using the default, the line [here](https://github.com/auth0/jwt-decode/blob/beta/lib/index.ts#L28) seems like dead-code. I want to leave that line because of people passing in null and not using anything that tells them it shouldnt.

By removing the default, that line is now also covered when passing in undefined (or nothing), making it no longer dead-code.

I don't mind adding in that change.

Reply:
Hmmm, yeah I see. The current types that are already in a published version also marked this as optional.

I guess this is a question of whether the library should do all this validation, or if that is the job of TypeScript. And if users don't use TypeScript, should we handle all these cases of input validation in code?

I'd argue that if users need to validate their input is correct, that they should adopt TypeScript. Especially considering we'd essentially be shipping more code to the user, even if they are running in production. I'd even argue we should remove the [existing checks](https://github.com/auth0/jwt-decode/blob/fa904b3adc167c2d75af381b807649ed74a2766e/lib/index.js#L13-L15) for specific types, and make a clean break to let TypeScript handle such things.

Reply:
Yup i'm also accustomed to rely on TypeScript or intellisense / code completion from the   editors. So as you say @jonkoops most people should get an warning should they ever try to pass null in an editor that uses intellisense and relies on the types.

However as @frederikprijck says here, there may be people out there that are passing null an won't get an warning when upgrading, should we had rely on the types only we should have then perhaps added an assert instead that checks if the passed input is null an throw an error instead.

This is an easy fix and we retain the same logic with the `options = options || {}` initializer as before.

Reply:
> Thie library didnt use TypeScript before v4, and we introduced TypeScript for convenience and easier tool alignment.
> 
> We don't want to make any assumptions on our users using typescript, that's not the scope of the new major.

Users that use an editor with intellisense / code completion that leverages typescript definitions file would have gotten an error/warning if they passed null as an argument does not matter if they are using TypeScript or not. Some editors have some basic type checking built in by default.